test/unit/shared/access_permissions_test_base.rb

```diff
@@ -21,10 +21,15 @@ def check_routes(app_routes, skipped_actions)
           controller = "#{r.defaults[:controller]}_controller".classify.constantize
           filters    = controller.send(:_process_action_callbacks)
 
-          # Pass if the controller deliberately only permit admins (e.g. SettingsController)
-          if filters.select { |f| f.filter == :require_admin }.empty?
-            assert_not_equal [], Foreman::AccessControl.permissions.select { |p| p.actions.include? path }
-          end
+          permission_required = true
```

Comment:
maybe just return early instead of having a variable? that way you wouldn't need to do the second check if the first already set the variable to `false`?

Reply:
hm... an early return gives me "LocalJumpError: unexpected return"... I guess we can't use that. :-(

Reply:
OK, figured it out - test are blocks, not functions, so to return early you need to use `next` instead of `return`

Reply:
Duh, makes sense. Thanks.